### DISCUSSION TITLE
onInputChange equivalent

### DISCUSSION THREAD

**Discussion by LightningManMedia** [dowjones/react-dropdown-tree-select#509] [Q&A]
Hi,

I was wondering if there's a function that could be used to handle when the input is changed by typing and return the text that has been typed before making a selection on the tree. I know that onChange is available but that seems to only apply to the selection made from the dropdown. 

I would appreciate any thoughts on this.

Thanks!
Chris

**Comment by mrchief** [comment]
@LightningManMedia what's your use case? onChange fires pretty frequently and the component debounces those. Plus all selection, tree filtering etc. is handled internally. If we had a handler like that, then with functional components and react in general, the component would re-render after every keystroke.

**Reply by LightningManMedia** [reply]
@mrchief Thank you for your response. The use case is to call an endpoint that passes in the value of the input field after each keystroke. It sounds like that isn't feasible due to the rendering overhead.

**Reply by mrchief** [reply]
Yeah. This is where React's whole "UI as a state" approach actually gets in the way. I guess with hooks and headless components this can be better solved but I haven't had time to work towards that.
